testsuite/testcases/src/public_fullnode_performance.rs

```diff
@@ -80,6 +82,31 @@ impl PFNPerformance {
 
         all_peer_ids
     }
+
+    /// Gathers and shuffles all peer IDs in the swarm, colocating VFNs with their validator
+    fn gather_and_shuffle_peer_ids_with_colocation(
+        &self,
+        swarm: &mut dyn Swarm,
+    ) -> Vec<Vec<AccountAddress>> {
+        // Identify the validators and fullnodes in the swarm
+        let validator_peer_ids = swarm.validators().map(|v| v.peer_id()).collect::<Vec<_>>();
+        let fullnode_peer_ids = swarm.full_nodes().map(|v| v.peer_id()).collect::<Vec<_>>();
+        let (vfn_peer_ids, pfn_peer_ids) =
+            fullnode_peer_ids.split_at(fullnode_peer_ids.len() - self.num_pfns as usize);
```

Comment:
Discussed offline with Brian: all the PFNs have index set to 0, so this split will not work.